loopy/kernel/__init__.py

```diff
@@ -1638,25 +1603,45 @@ def __ne__(self, other):
 
     # }}}
 
+    def get_copy_kwargs(self, **kwargs):
+        if "iname_to_tags" in kwargs:
+            assert "inames" not in kwargs
+            iname_to_tags = kwargs["iname_to_tags"]
+            domains = kwargs.get("domains", self.domains)
+            kwargs["inames"] = {name: Iname(name,
+                                            iname_to_tags.get(name, frozenset()))
+                                for name in _get_inames_from_domains(domains)
+                                }
+            del kwargs["iname_to_tags"]
```

Comment:
This needs to do a better job merging the existing `self.inames` with new information:
- Existing `inames` might already have tags. It seems as though this blows those away?
- Should we try merging existing tags and new tags even for inames for which `iname_to_tag[iname]` is provided? I suspect "no" is the safer answer.

Reply:
Duplicate of https://github.com/inducer/loopy/pull/293#discussion_r648850082.

Reply:
You're right, passing `iname_to_tags` amounts to wholesale replacement. This is OK.